### PR TITLE
Expose cdr

### PR DESCRIPTION
**Pull request by Karsten1987**
Connects to ros2/demos#185

### CODE REVIEW

File: test_communication/test/test_publisher_subscriber_raw.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  void TearDown()
+  {
+    rclcpp::shutdown();
+  }
+};
+
+TEST_F(CLASSNAME(TestMessageSerialization, RMW_IMPLEMENTATION), raw_callback) {
+  size_t counter = 0;
+
+  auto raw_callback = [&counter](const std::shared_ptr<rmw_message_raw_t> raw_msg) {
```
Comment (sloretz) [inline test_communication/test/test_publisher_subscriber_raw.cpp]: It looks like this test will pass if `raw_callback` is never called. Maybe add `EXPECT_GT(counter, 0u)` after the loop at the bottom?

File: test_communication/test/test_message_serialization.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  auto raw_message_c = rmw_get_zero_initialized_raw_message();
+  rmw_initialize_raw_message(&raw_message_c, 0, &allocator);
+
+  auto message_c_typesupport = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BoundedArrayNested);
+  test_msgs__msg__BoundedArrayNested bounded_array_nested_msg_c;
+  fill_c_message(&bounded_array_nested_msg_c);
+
+  auto ret = RMW_RET_OK;
+  ret =
+    rmw_serialize(&bounded_array_nested_msg_c, message_c_typesupport, &raw_message_c);
```
Comment (sloretz) [inline test_communication/test/test_message_serialization.cpp]: It looks like this call would fit on the line above

File: test_communication/test/test_message_serialization.cpp [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  test_msgs::msg::BoundedArrayNested msg_cpp;
+  fill_cpp_message(&msg_cpp);
+
+  auto ret = rmw_serialize(&msg_c, message_c_typesupport, &raw_message_c);
+  ASSERT_EQ(RMW_RET_OK, ret);
+  ret = rmw_serialize(&msg_cpp, message_cpp_typesupport, &raw_message_cpp);
+  EXPECT_EQ(RMW_RET_OK, ret);
+
+  print_raw_buffer(raw_message_c, "raw message c");
+  print_raw_buffer(raw_message_cpp, "raw message cpp");
```
Comment (sloretz) [inline test_communication/test/test_message_serialization.cpp]: Are `raw_message_c.buffer` and `raw_message_cpp.buffer` expected to be identical?

**Comment by Karsten1987** [comment]
but the static typesuppport branch has conflicts. Most likely due to my merge. 
I believe the branch has to be rebased:
https://github.com/ros2/rmw_fastrtps/pull/203

**Pull request merged**